core/src/main/java/com/linecorp/armeria/server/cors/CorsPolicy.java

```diff
@@ -0,0 +1,285 @@
+     * caching of the preflight response for the specified time. During this time no preflight
+     * request will be made.
+     *
+     * @return the time in seconds that a preflight request may be cached.
+     */
+    public long maxAge() {
+        return maxAge;
+    }
+
+    /**
+     * Returns a set of headers to be exposed to calling clients.
+     *
+     * <p>During a simple CORS request only certain response headers are made available by the
+     * browser, for example using:
+     * <pre>{@code
+     * xhr.getResponseHeader("Content-Type");
+     * }</pre>
+     * The headers that are available by default are:
+     * <ul>
+     * <li>Cache-Control</li>
+     * <li>Content-Language</li>
+     * <li>Content-Type</li>
+     * <li>Expires</li>
+     * <li>Last-Modified</li>
+     * <li>Pragma</li>
+     * </ul>
+     *
+     * <p>To expose other headers they need to be specified, which is what this method enables by
+     * adding the headers names to the CORS 'Access-Control-Expose-Headers' response header.
+     *
+     * @return the list of the headers to expose.
+     */
+    public Set<AsciiString> exposedHeaders() {
+        return exposedHeaders;
+    }
+
+    /**
+     * Returns the allowed set of Request Methods. The Http methods that should be returned in the
+     * CORS 'Access-Control-Request-Method' response header.
+     *
+     * @return the {@link HttpMethod}s that represent the allowed Request Methods.
+     */
+    public Set<HttpMethod> allowedRequestMethods() {
+        return allowedRequestMethods;
+    }
+
+    /**
+     * Returns the allowed set of Request Headers.
+     *
+     * <p>The header names returned from this method will be used to set the CORS
+     * 'Access-Control-Allow-Headers' response header.
+     */
+    public Set<AsciiString> allowedRequestHeaders() {
+        return allowedRequestHeaders;
+    }
+
+    /**
+     * Determines if the policy allows a null origin.
+     */
+    public boolean isNullOriginAllowed() {
+        return nullOriginAllowed;
+    }
+
+    /**
+     * Returns HTTP response headers that should be added to a CORS preflight response.
+     *
+     * @return {@link HttpHeaders} the HTTP response headers to be added.
+     *
+     * @throws IllegalStateException if CORS support is not enabled
+     */
+    public HttpHeaders preflightResponseHeaders() {
+        if (preflightResponseHeaders.isEmpty()) {
+            return HttpHeaders.EMPTY_HEADERS;
+        }
+
+        final HttpHeaders preflightHeaders = new DefaultHttpHeaders(false);
+        for (Entry<AsciiString, Supplier<?>> entry : preflightResponseHeaders.entrySet()) {
+            final Object value = getValue(entry.getValue());
+            if (value instanceof Iterable) {
+                preflightHeaders.addObject(entry.getKey(), (Iterable<?>) value);
+            } else {
+                preflightHeaders.addObject(entry.getKey(), value);
+            }
+        }
+        return preflightHeaders;
```

Comment:
Could we build an immutable `HttpHeaders` and reuse it, instead of building it every time?